source/agora/consensus/EnrollmentManager.d

```diff
@@ -751,6 +760,55 @@ public class EnrollmentManager
         }
     }
 
+    /***************************************************************************
+
+        Get the cycle index of `PreImageCycle`
+
+        Returns:
+            the cycle index
+
+    ***************************************************************************/
+
+    private uint getCycleIndex () @trusted nothrow
+    {
+        try
+        {
+            auto results = this.db.execute("SELECT val " ~
+                "FROM node_enroll_data where key = ?", "cycle_index");
+            if (results.empty)
+                return 0;
+
+            return results.oneValue!(uint);
+        }
+        catch (Exception ex)
+        {
+            log.error("ManagedDatabase operation error {}", ex);
+            return 0;
```

Comment:
Will the calling code properly handle this condition?

Reply:
It's a proper question. But, It's a difficult problem. If there's a problem with the failure of restoring the cycle index, what should we do? The only way to restore the cycle index is to search for all the enrollment from the Genesis block to the lastest block. We can find a right solution for that if we research on the issue #849  I guess. Anyway, I have two ideas about what you are questioning. What do you think is better for that between just adding `assert(0)` or restoring it from looking into all the blocks at the moment?

Reply:
We need to get some better error information from sqlite before we can do proper error handling (see for example https://github.com/bpfkorea/d2sqlite3/pull/1).

For now we could proceed with this try/catch code. Later we will have to fix our usage of sqlite.

Reply:
> We need to get some better error information from sqlite before we can do proper error handling (see for example bpfkorea/d2sqlite3#1).

I'll take a look at the PR. Thanks.